### ISSUE TITLE
Seg Fault on CGO3+ JPG Extracting Features

### ISSUE THREAD

**Issue by charlescva** [sweeneychris/TheiaSfM#170]
feature_extractor.h 82: bool ExtractToDisk(const std::vector<std::string>& filenames);
Has segmentation fault when using JPG from this camera. http://us.yuneec.com/cgo3-plus-overview

Running jhead produces:

> Nonfatal Error : '../reconstructions/Simeon House/YUN00001.jpg' Extraneous 173 padding bytes before section DB
> File name    : ../reconstructions/Simeon House/YUN00001.jpg
> File size    : 2658040 bytes
> File date    : 2017:07:08 06:42:22
> Camera make  : YUNEEC
> Camera model : CGO3P 3.2.26(A)
> Date/Time    : 2017:07:08 10:42:21
> Resolution   : 3840 x 2160
> Exposure time: 0.0041 s  (1/243)
> Aperture     : f/2.8
> ISO equiv.   : 100
> Whitebalance : Auto
> Metering Mode: average
> Exposure     : program (auto)
> GPS Latitude : ...
> GPS Longitude: ...
> GPS Altitude : ...
> JPEG Quality : 80

Image looks fine with gnome built-in image viewer, so as the error above is correct that it is nonfatal.  Never had this issue with any other jpeg, and already emailed Yuneec asking about it as well.  But wanted to share so others who have similar issues can have a data point to go on.

I tried looking through the code a bit, but I didnt see anything exif related, so maybe its a FloatImage thing?  Will update this whenever I get to the bottom of the issue.

**Comment by sweeneychris** [comment]
This seems to be something weird with OpenImageIO not being able to read the image. I don't think there's much to do in Theia to fix this since that's an external dependency. One thing you can try is just to re-save the images to enforce them to be rewritten/re-encoded. Perhaps with an [imagemagick](https://www.imagemagick.org/script/index.php) library command like mogrify -resize 100% so that the pixels are not actually altered.

**Comment by charlescva** [comment]
Yeah I did that and it worked out pretty well so it is a short fix for the
time being. I know there's a ticket already about OpenImageIO that would
address this issue if we had a feasible replacement.  Will do some more
investigating on my end and update as information is available.

**Comment by charlescva** [comment]
I ran these images through a GIMP batch script, and Gimp also complained about the extraneous 173 bytes.  Additionally it complained about the XSD schema within the metadata.  Yuneec emailed me back and simply stated that the data is metadata for the JPEG, and that anything specific about the camera is proprietary to Yuneec, blah blah blah.

Closing this issue because I think it's just poor standards compliance on the vendor's camera, unfortunately.
